src/msgraph_core/models/page_result.py

```diff
@@ -0,0 +1,127 @@
+"""
+This module defines the PageResult class which represents a page of
+items in a paged response.
+
+The PageResult class provides methods to get and set the next link and
+the items in the page, create a PageResult from a discriminator value, set
+the value, get the field deserializers, and serialize the PageResult.
+
+Classes:
+    PageResult: Represents a page of items in a paged response.
+"""
+from typing import Any, List, Optional
+
+from kiota_abstractions.serialization.parsable import Parsable  # type: ignore
+from kiota_abstractions.serialization.serialization_writer \
+     import SerializationWriter  # type: ignore
+from kiota_abstractions.serialization.parse_node import ParseNode  # type: ignore
+from typing import TypeVar, List, Optional, Generic
+
+T = TypeVar('T')
+
+
+class PageResult(Parsable, Generic[T]):
```

Comment:
[suggestion] Since this is `Parsable`  it should follow the format in https://github.com/microsoft/kiota-abstractions-python/blob/main/kiota_abstractions/serialization/parsable.py and be a `dataclass` instance. Then we can also do away with getters and setters and use the dataclass fields like so

```py
@dataclass
class PageResult(Parsable, Generic[T]):
    odata_next_link: Optional[str] = None
    value: Optional[List[Parsable]] = None
```